### ISSUE TITLE
Problem with private argument

### ISSUE THREAD

**Issue by elgow** [evanunderscore/defopt#50]
I'm getting an exception when running  a function with a keyword-only private argument.  In this case the arg is called "_sh".

  File ".../defopt.py", line 352, in _call_function
    arg = getattr(args, name)
AttributeError: 'Namespace' object has no attribute '_sh'

In the code this is what's happening:

```
    347 def _call_function(func, args):
    348     positionals = []
    349     keywords = {}
    350     sig = _inspect_signature(func)
    351     for name, param in sig.parameters.items():
--> 352         arg = getattr(args, name)
    353         if param.kind in [param.POSITIONAL_ONLY, param.POSITIONAL_OR_KEYWORD]:
    354             positionals.append(arg)
    355         elif param.kind == param.VAR_POSITIONAL:
    356             positionals.extend(arg)
    357         else:
```
name = '_sh'
param = <Parameter "_sh=<class 'sh.Command'>">
args = Namespace(_func=<function git_repo_info at 0x10443de18>, raw=False, remote='origin', verbose=True, working_dir='.')

So the default value for the argument `_sh` is present in the value of  the`param` object, but it's not being picked up. It seems that what should happen is to check if the param has a default value and change line 352 to 

```
arg = getattr(args, name, param.default)
```

**Comment by evanunderscore** [comment]
Thanks for reporting! Discussion continues on #51.

**Comment by evanunderscore** [comment]
Fixed by #53.
